### ISSUE TITLE
bracketed paste mode needs a time out

### ISSUE THREAD

**Issue by ailin-nemui** [irssi/irssi#942]
```
unlord_> hi
unlord_> I have irssi running in a screen session
Soliton> well done.
unlord_> when I attach, I can see that irssi is updating, the clock is ticking, notifications light up windows, etc. but the keyboard does not work
unlord_> nothing I type goes through to irssi
Nei> what fixed it?
unlord_> 08:43 <+Nei> try blindly typing "Esc" "[" "2" "0" "1" "~"
```

**Comment by tejeez** [comment]
The same thing happened to me after pasting an URL in irssi. An extra ~ appeared after the text and irssi stopped responding to keyboard.

That \e[201~ sequence fixed it for me after keyboard input had been stuck for several days.

**Comment by ailin-nemui** [comment]
did you try weechat any time before this occurred?

**Comment by tejeez** [comment]
> did you try weechat any time before this occurred?

No, I didn't. Could some other program have changed my terminal settings somehow? Irssi was running in tmux though, so I first thought it might be a problem with tmux, but apparently it had happened with screen as well.

(I hadn't actually even heard of bracketed paste mode before but somehow managed to find the right issue...)

**Comment by bw1** [comment]
[ctlseqs](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-Bracketed-Paste-Mode)

## Bracketed Paste Mode

When bracketed paste mode is set, pasted text is bracketed with control
sequences so that the program can differentiate pasted text from typed-
in text.  When bracketed paste mode is set, the program will receive:
   ESC [ 2 0 0 ~ ,
followed by the pasted text, followed by
   ESC [ 2 0 1 ~ .

```
CSI ? Pm h          DEC Private Mode Set (DECSET).
Ps = 2 0 0 4  -> Set bracketed paste mode, xterm.
CSI ? Pm l           DEC Private Mode Reset (DECRST).
Ps = 2 0 0 4  -> Reset bracketed paste mode, xterm.
```

[strcat](http://www.strcat.de/blog/archives/1775-Vim,-Tmux,-bracketed-paste-mode,-italics-und-Easteregg.html) <-german

**Comment by ailin-nemui** [comment]
it is at least likely that some other program changed your terminal mode, and did not change it back, thus exposing this Irssi bug
